src/rebar_packages.erl

```diff
@@ -91,9 +130,62 @@ handle_missing_package(Dep, State, Fun) ->
             throw(?PRV_ERROR({missing_package, Dep}))
     end.
 
-registry_dir(State) ->
+signature(Registry) ->
+    case rebar_utils:url_append_path(Registry, ?REMOTE_REGISTRY_FILE ++ ".signed") of
+        {ok, Url} ->
+            ?DEBUG("Fetching signature from ~p", [Url]),
+            case httpc:request(get, {Url, [{"User-Agent", rebar_utils:user_agent()}]},
+                               [], [{sync, true}, {body_format, binary}], rebar) of
+                {ok, {{_, 200, _}, _, Signature}} ->
+                    {ok, list_to_binary(decode_binary(Signature))};
+                _ ->
+                    signature_fetch_fail
+            end;
+        _ ->
+            signature_fetch_fail
+    end.
+
+decode_binary(Binary) ->
+    [erlang:binary_to_integer(<<C:16>>, 16) || <<C:16>> <= Binary].
+
+-spec public_key(string(), rebar_state:t()) -> {ok, binary()} | {error, atom()}.
+public_key(Registry, State) ->
+    {ok, RegistryDir} = rebar_packages:registry_dir(Registry, State),
+    File = filename:join(RegistryDir, "cert.pem"),
+    case file:read_file(File) of
+        {error, _} when Registry =:= ?DEFAULT_CDN ->
+            %% If we are using the default registry we can fall back to
+            %% the public key included within rebar3 itself.
+            {ok, ?DEFAULT_REPO_PUBLIC_KEY};
+        Result ->
+            Result
+    end.
+
+maybe_verify_registry(Registry, Data, State) ->
+    maybe_verify_registry(os:getenv(?HEX_UNSAFE_REGISTRY), Registry, Data, State).
+
+maybe_verify_registry(false, Registry, Data, State) ->
```

Comment:
Oh, I get what you mean. Hm, I've usually just relied on the user having to `unset` if they want something turned off. I suppose we could do `when Unsafe =:= false ; Unsafe =:= "false"` just in case?